src/GraphQLParser/LexerContext.cs

```diff
@@ -116,94 +101,163 @@ public Token ReadNumber()
             return isFloat ? CreateFloatToken(start) : CreateIntToken(start);
         }
 
+        public Token ReadComment()
+        {
+            int start = _currentIndex;
+            char code = NextCode();
+
+            Span<char> buffer = stackalloc char[4096];
+            StringBuilder? sb = null;
+
+            int index = 0;
+            bool escaped = false;
+
+            while (_currentIndex < _source.Length && code != 0x000A && code != 0x000D)
+            {
+                char ch = ReadCharacterFromString(code, ref escaped);
+
+                try
+                {
+                    buffer[index++] = ch;
+                }
+                catch (IndexOutOfRangeException) // fallback to StringBuilder in case of buffer overflow
+                {
+                    if (sb == null)
+                        sb = new StringBuilder(buffer.Length * 2);
+
+                    for (int i = 0; i < buffer.Length; ++i)
+                        sb.Append(buffer[i]);
+
+                    sb.Append(ch);
+                    index = 0;
+                }
+
+                code = NextCode();
+            }
+
+            if (sb != null)
+            {
+                for (int i = 0; i < index; ++i)
+                    sb.Append(buffer[i]);
+            }
+
+            var value = escaped
+                ? (sb == null ? buffer.Slice(0, index).ToString() : sb.ToString()).AsMemory() // allocate string from either buffer on stack or heap
+                : _source.Slice(start + 1, _currentIndex - start - 1); // the best case, no escaping so no need to allocate
+
+            return new Token
+            (
+                TokenKind.COMMENT,
+                value,
+                start,
+                _currentIndex + 1
+            );
+        }
+
         public Token ReadString()
         {
             int start = _currentIndex;
             char code = NextCode();
-            var sb = new StringBuilder(); // TODO: buffer allocation; it can be pooled later
 
-            while (IsNotAtTheEndOfQuery() && code != 0x000A && code != 0x000D && code != '"')
+            Span<char> buffer = stackalloc char[4096];
+            StringBuilder? sb = null;
```

Comment:
Then the same stringbuilder can be used here.

Reply:
see https://github.com/graphql-dotnet/parser/pull/101#discussion_r554628143